pep-0581.rst

```diff
@@ -0,0 +1,362 @@
+
+Currently the CLA process involves real human to manually check the records.
+Question: Will it be possible to completely automate the CLA process, so
+it does not require human intervention?
+
+Create labels for issue triage
+------------------------------
+
+In bpo, we currently have the following fields for each issues:
+
+Type: behavior, crash, compile error, resource usage, security, performance, enhancement.
+Components: 2to3, Argument Clinic, asyncio, Build, Cross-build, ctypes, ...
+Priority: release blocker, deferred blocker, critical, high, normal, low
+
+We will create the corresponding labels::
+
+   type-behavior, type-crash, type-compile error, type-resource usage, ...
+
+   components-2to3, components-argument clinic, components-asyncio, ...
+
+   priority-release blocker, priority-deferred blocker, priority-critical, ...
+
+In addition, we'll create ``needs triage`` label.
+
+Create issue template
+---------------------
+
+We will create an issue template and add the following headers::
+
+   ---
+   Type: behavior | crash | compile error | resource usage (choose one)
+   Components: 2to3 | Argument Clinic | asyncio | Build | ... (can select more than one)
+   Priority: release blocker | deferred blocker | critical | ...
+   Needs backport to: 2.7 | 3.6 | 3.7
+   ---
+
+The idea is to allow the issue creator to help us triage the issue.
+The above values are pre-filled in the template. Issue creator will remove texts
+that do not apply to their issue.
+
+Based on the above headers, bedevere-bot can apply the necessary labels to the
+issue. If issue creator did not supply the above headers, the bot will apply
+``needs triage`` label. At that point it will require a core developer to
+properly label the issue.
+
+Updates to bedevere
+-------------------
+
+Bedevere-bot will need to be updated to recognize the issue headers described above,
+and apply the proper labels.
+
+Bedevere-bot can also copy over the labels to pull requests that correspond to
+the issue.
+
+Update the devguide
+-------------------
+
+Provide explanation in the devguide about new issue workflow and triage labels.
+
+Add a button in bpo to migrate the issue to GitHub
+--------------------------------------------------
+
+This will require actual update to the bpo. But I believe this effort needed
+is much less than a complete overhaul.
+
+We will create a button in bpo, that will copy over the issue description
+and associated comments into a GitHub issue.
+
+We should not be moving all open issues to GitHub. Issues with little or no
+activity should just be closed. Issues with no decision made for years should
+just be closed.
+
+If a core developer is still interested in the issue, they can indicate so in
+the bpo issue, and later use the button to migrate it over to GitHub.
+
+Make bpo readonly
+-----------------
+
+This should be the final step. Once we start using GitHub issues, make bpo
+readonly, not shut it down.
```

Comment:
It was not practical to use hg and git simultaneously.  In my view, and some others, apparently, github was not fully ready to use when the switch was made.  I would say not until Miss Islington started doing backports.  (They still take more time than on hg, but the difference is tolerable.)  The result was that many people stopped patching python for various periods.

It *would* be practical to have both issue trackers in use for a short time.  So I think disabling bpo should wait until the new system has been used and approved by people other than the creators.  Unlike with the repository, testing can be done with real issues for cpython PRs.

So I think approval of this PEP should be provisional, subject to final acceptance review, which would then signal 'stop using bpo'.  At that point, shut off new registrations and issues, (which could be discourage earlier), and new comments.  'Readonly' should mean that search still works, and that issues can still be closed instead of transferred.